test/unit/effects.js

```diff
@@ -1850,6 +1852,35 @@ test("Animation callbacks (#11797)", 15, function() {
 	this.clock.tick( 10 );
 });
 
+test("Animation callbacks (#2292)", 1, function() {
+	var jq_elem = jQuery("#foo"),
+		dur = 50,
+		cb = "";
+
+	jq_elem.animate( {
+		width: "5px"
+	}, {
+		duration: dur,
+		start: function() {
+			cb += "[s]";
+		},
+		progress: function( anim, p, ms ) {
+			cb += "[p," + p + "," + ms + "]";
+		},
+		done: function() {
+			cb += "[d]";
+		},
+		fail: function() {
+			cb += "[f]";
+		},
+		always: function() {
+			equal( cb, "[s][p,0," + dur + "][p,1,0][d]", "callbacks order" );
```

Comment:
This is also too opaque. How about pulling in [qunit-assert-step](https://github.com/JamesMGreene/qunit-assert-step)?